gradle.properties

```diff
@@ -6,7 +6,7 @@ kotlin.native.ignoreDisabledTargets=true
 kotlin.mpp.enableCompatibilityMetadataVariant=true
 
 # SDK
-sdkVersion=0.16.6-SNAPSHOT
+sdkVersion=0.17.0-SNAPSHOT
```

Comment:
correctness: since this is a breaking change, we should have an [exclamation point in the PR title](https://github.com/awslabs/smithy-kotlin/blob/main/CONTRIBUTING.md#footer-optional). also maybe add `**Breaking**` somewhere in the changelog

will we have a discussion post to help users unbreak themselves?